post-to-aws/src/test/java/S3StorageImplTest.java

```diff
@@ -145,4 +151,33 @@ var record = new Record();
     private Record createNullRecord() {
         return new Record();
     }
+
+    private static final class FakeMultipartUploadS3Client extends FakeS3Client {
+
+        private final List<String> keys;
+
+        private FakeMultipartUploadS3Client() {
+            this.keys = new ArrayList<>();
+        }
+
+        @Override
+        public CompleteMultipartUploadResponse completeMultipartUpload(CompleteMultipartUploadRequest request) {
+            keys.add(request.key());
+            return CompleteMultipartUploadResponse.builder().build();
+        }
+
+        @Override
+        public CreateMultipartUploadResponse createMultipartUpload(CreateMultipartUploadRequest request) {
+            return CreateMultipartUploadResponse.builder().uploadId(randomString()).build();
+        }
+
+        @Override
+        public UploadPartResponse uploadPart(UploadPartRequest request, RequestBody requestBody) {
+            return UploadPartResponse.builder().eTag(randomString()).build();
+        }
+
+        public String getSingleItem() {
+            return keys.get(0);
```

Comment:
Not on modern Java?

Reply:
Nope, Java11, the version we have installed on all brage instances